v2/pkg/protocols/common/executer/executer.go

```diff
@@ -15,6 +15,10 @@ import (
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/helpers/writer"
 )
 
+// ShowFailureMatchPerRequest if set to true, will show failure match per request
+// by defalut shows failure match per template.
+var ShowFailureMatchPerRequest bool
+
 // Executer executes a group of requests for a protocol
```

Comment:
@RamanaReddy0M requesting minor change regarding implementation . we can move this variable to types.Options and access it here in Executor with `e.options.Options.MatchStatusPerRequest`

also this variable seems to  be unused and hence always false

- [ ] add new flag `-msr -matcher-status-request` to show matcher status per request (use ^ variable)